nifpga/session.py

```diff
@@ -92,14 +93,17 @@ def __init__(self,
         if no_run:
             open_attribute = open_attribute | OPEN_ATTRIBUTE_NO_RUN
 
-        bitfile_path = bytes(bitfile.filepath, 'ascii')
-        bitfile_signature = bytes(bitfile.signature, 'ascii')
-        resource = bytes(resource, 'ascii')
-        self._nifpga.Open(bitfile_path,
-                          bitfile_signature,
-                          resource,
-                          open_attribute,
-                          self._session)
+        if isinstance(resource, _SessionType):
```

Comment:
Whats the goal with reusing the underlying session?  When the first Session closes it will close the _session and break the second Session.

Why not just create another reference to the original Session?

Reply:
Talked offline, basically it opens the door for potential integration with other APIs in the future.